hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -379,6 +374,15 @@ def calls(self):
         test function."""
         return self.engine.call_count
 
+    def consider_new_tree(self, tree):
+        data = ConjectureData.for_ir_tree(tree)
+        self.engine.test_function(data)
+
+        data = ConjectureData.for_buffer(data.buffer)
+        self.engine.test_function(data)
+
+        return self.consider_new_buffer(data.buffer)
+
```

Comment:
this triples our shrink cost...we could feasibly get it down to double, since the second pass is for maintaining the correct `forced` values (which affects `block.trivial`). It's just annoying and messy to thread a `fake_forced` parameter through everywhere.

At some point, more than half of our passes will be on the ir, at which point it's more efficient to convert from bits to ir when necessary, and maintain the tree as the source of truth. So in practice, this caps out at 1.75x slowdown or so at the inflection point.

Reply:
I think that's worth it; 3x is a very heavy cost to pay.

Could we also drop the cost of this first stage by only running the IR logic if there's a float in the tree?  That way there's no slowdown (well, small constant) for unaffected tests.

Reply:
I think that's already the case: `consider_new_tree` is only called when `Float.shrink` runs, which is gated behind a `leaf = chooser.choose(self.leaves, lambda leaf: leaf.ir_type == "float")`.

Reply:
Oh, nice, if that's worst-case instead of average it's not such a big deal; hopefully also offset by high quality/specificity shrinking.

Reply:
Well, good news is I was being foolish / overzealous and the extra third call isn't necessary 😄. We're dealing purely with the buffer, so who cares about the forced status of the blocks.